terracotta/server/metadata.py

```diff
@@ -50,6 +50,35 @@ class Meta:
     )
 
 
+class CommaSeparatedListField(fields.Field):
+    def _deserialize(self, value, attr, data, **kwargs):
+        try:
+            assert value[0] == "[" and value[-1] == "]"
+
+            if value == "[]":
+                return []
+
+            return value[1:-1].split(", ")
+        except ValueError:
+            raise ValidationError("Invalid input for a list of values.")
+
+
+class MetadataColumnsSchema(Schema):
+    columns = CommaSeparatedListField(description="Columns of dataset to be returned")
```

Comment:
I think `fields.List(fields.String())` will do here, should be easy enough to jsonify that on the frontend.

Reply:
Will do. I was unsure if you wanted the quotations or not (based on the example provided) - but that makes more sense :)

Reply:
Actually, it doesn't seem to be that simple 😦 

All the queries are strings, so we'd have to parse this string `["metadata", "bounds"]`. I'm not really sure what would be the best way to tackle this.

We could use JSON parsing or `ast.literal_eval` - but they both seem a bit unideal. Each solution requires its own field, anyway - unless we want to add this functionality to the handler itself (which I personally don't think we should do).

I'll wait for your advice before progressing. Perhaps I'm missing something obvious.

Reply:
I don't get it. `fields.List(fields.String())` should work just fine on requests like this:

```
POST /metadata?columns=["metadata","bounds"]
```

Reply:
`POST http://localhost:5000/metadata?columns=["metadata", "bounds"]`

```py
class MetadataColumnsSchema(Schema):
    columns = fields.List(
        fields.String(),
        description="List of columns to return",
        required=False,
    )
```
Handler:
```py
columns_schema = MetadataColumnsSchema()
columns = columns_schema.load(request.args)
```

Calling that raises the following exception:
```py
marshmallow.exceptions.ValidationError: {'columns': ['Not a valid list.']}
```
This makes sense, because: 
```py
print(request.args)
print(request.args.get("columns"))
print(type(request.args.get("columns")))
```
prints the following
```
ImmutableMultiDict([('columns', '["metadata", "bounds"]')])
["metadata", "bounds"]
<class 'str'>
```
---
We could go with using JSON to parse the string first into a list or `ast`. What would you prefer? Something along these lines:
```py
columns_str = request.args.get("columns")
try:
    columns_list = json.loads(columns_str)
except json.JSONDecodeError:
    ...
    
columns = columns_schema.load({"columns": columns_list})
```
or
```py
class JSONStringListField(fields.Field):
    def _deserialize(self, value, attr, data, **kwargs):
        try:
            return json.loads(value)
        except json.JSONDecodeError:
            raise ValidationError("Invalid JSON string format.")
```
What do you think?

Reply:
I see. Other routes handle this via JSON: 

https://github.com/DHI-GRAS/terracotta/blob/3b28c08852e66d9cceeb34ed6c458f934d36e2d6/terracotta/server/colormap.py#L46-L58

You can just copy this pattern.

Reply:
Cheers! It works as expected now 😄